src/Backend/Modules/Blog/Engine/Model.php

```diff
@@ -440,24 +440,27 @@ public static function getCategory($id): array
     }
 
     /**
-     * Get a category id by title
-     *
-     * @param string $title    The title of the category.
-     * @param string $language The language to use, if not provided we will use the working language.
-     *
-     * @return int
+     * @deprecated
      */
-    public static function getCategoryId(string $title, string $language = null): int
+    public static function getCategoryId(string $title, ?string $language): int
     {
-        $title = (string) $title;
-        $language = ($language !== null) ? (string) $language : BL::getWorkingLanguage();
+        if ($language === null) {
+            $language = BL::getWorkingLanguage();
+        }
 
-        return (int) BackendModel::getContainer()->get('database')->getVar(
-            'SELECT i.id
-             FROM blog_categories AS i
-             WHERE i.title = ? AND i.locale = ?',
-            [$title, $language]
-        );
+        $categories = BackendModel::get('blog.repository.category')
+                                  ->findBy(
+                                      [
+                                          'title' => $title,
+                                          'locale' => $language,
```

Comment:
`Locale::fromString($language ?? BL::getWorkingLanguage())`
And then you can drop the if statement

You need to use the Locale here because it isn't a string but a Locale value object

Reply:
Fixed in 5762dc28fd712cccabfb1dad7fdf4173ae61b7e4